qt/widgets/instrumentview/src/InstrumentActor.cpp

```diff
@@ -1058,8 +1059,8 @@ void InstrumentActor::setDataIntegrationRange(const double &xmin,
       // Ignore monitors if they are masked on the view
       if (spectrumDefinition.size() == 1 &&
           (std::find(monitorIndices.begin(), monitorIndices.end(), i) !=
-               monitorIndices.end() ||
-           maskWksp->isMasked(static_cast<int>(i))))
+               monitorIndices.end() || (maskWksp != nullptr &&
```

Comment:
Works for me, the only nit tweaking this to get rid of the comparison then it matches the common usage across the codebase:

`(maskWksp && maskWksp->isMasked(...))`

Reply:
Thank you for having a look into this. I removed the comparison and also modified the warning message when the ExtractMask algorithm fails to include information that scanning workspaces are not supported yet, as suggested by @gvardany .